src/Sylius/Component/Product/spec/Model/ProductSpec.php

```diff
@@ -176,10 +176,9 @@ function its_master_variant_should_be_mutable_and_define_given_variant_as_master
     function it_should_not_add_master_variant_twice_to_collection(VariantInterface $variant)
     {
         $variant->isMaster()->willReturn(true);
-        $variant->isDeleted()->willReturn(false);
 
-        $variant->setProduct($this)->shouldBeCalled();
-        $variant->setMaster(true)->shouldBeCalled();
+        $variant->setProduct($this)->shouldBeCalledTimes(1);
+        $variant->setMaster(true)->shouldBeCalledTimes(2);
```

Comment:
It is strongly related to `it_should_not_add_master_variant_twice_to_collection`, because we check if after setting first master variant adding second one is possible. As I wrote before it helps me to find a bug in my changes. 

Moreover if you would add two variants this spec will fail, because both of them would be in variants collection. But if we are talking about master variants it was possible to add two master variants, and pass this tests. That's why I propose this change. Right now, if something similar would happen this test will fail, because on both master variants `setProduct` would be call, but it shouldn't.